### DISCUSSION TITLE
snmp_exporter - generator error (config.go:27:18: undefined: os.ReadFile)

### DISCUSSION THREAD

**Discussion by vjanandd** [prometheus/snmp_exporter#793] [General]
can some one help on the below error

root@prometheus:~# go get github.com/prometheus/snmp_exporter/generator
# github.com/prometheus/snmp_exporter/config
go/src/github.com/prometheus/snmp_exporter/config/config.go:27:18: undefined: os.ReadFile
root@prometheus:~#

**Comment by xprnvd** [comment]
I was able to get it working by using importing "io/ioutil"

During this incremental process, I had to change a couple of files. For example here's how my go/src/github.com/prometheus/snmp_exporter/config/config.go change looks

```go

package config

import (
        "fmt"
        "regexp"
        "time"
        "io/ioutil"

        "github.com/gosnmp/gosnmp"
        "gopkg.in/yaml.v2"
)

func LoadFile(filename string) (*Config, error) {
        content, err := ioutil.ReadFile(filename)
        if err != nil {
                return nil, err
        }
        cfg := &Config{}
        err = yaml.UnmarshalStrict(content, cfg)
        if err != nil {
                return nil, err
        }
        return cfg, nil
}
```

Before taking this route, I would suggest checking the installed `go version` which for me was 1.12.9

This discussion may also help https://github.com/go-critic/go-critic/issues/1019

**Comment by SuperQ** [comment]
`io/ioutil` is deprecated by Go. Go 1.17 is required for building this code.
